Assets/VRTK/SDK/HyperealVR/SDK_HyperealVRHeadset.cs

```diff
@@ -0,0 +1,122 @@
+        public override Transform GetHeadsetCamera()
+        {
+            GameObject myHeadsetCameraGO = VRTK_SharedMethods.FindEvenInactiveGameObject<HyCamera>();
+            cachedHeadsetCamera = myHeadsetCameraGO != null ? myHeadsetCameraGO.transform : null;
+            return cachedHeadsetCamera;
+        }
+
+        /// <summary>
+        /// The GetHeadsetVelocity method is used to determine the current velocity of the headset.
+        /// </summary>
+        /// <returns>A Vector3 containing the current velocity of the headset.</returns>
+        public override Vector3 GetHeadsetVelocity()
+        {
+            HyTrackingState headsetState = HyperealVR.Instance.GetTrackingState(HyDevice.Device_HMD0);
+            return headsetState.velocity;
+        }
+
+        /// <summary>
+        /// The GetHeadsetAngularVelocity method is used to determine the current angular velocity of the headset.
+        /// </summary>
+        /// <returns>A Vector3 containing the current angular velocity of the headset.</returns>
+        public override Vector3 GetHeadsetAngularVelocity()
+        {
+            HyTrackingState headsetState = HyperealVR.Instance.GetTrackingState(HyDevice.Device_HMD0);
+            return headsetState.angularVelocity;
+        }
+
+        /// <summary>
+        /// The HeadsetFade method is used to apply a fade to the headset camera to progressively change the colour.
+        /// </summary>
+        /// <param name="color">The colour to fade to.</param>
+        /// <param name="duration">The amount of time the fade should take to reach the given colour.</param>
+        /// <param name="fadeOverlay">Determines whether to use an overlay on the fade.</param>
+        public override void HeadsetFade(Color color, float duration, bool fadeOverlay = false)
+        {
+            VRTK_ScreenFade.Start(color, duration);
+        }
+
+        /// <summary>
+        /// The HasHeadsetFade method checks to see if the given game object (usually the camera) has the ability to fade the viewpoint.
+        /// </summary>
+        /// <param name="obj">The Transform to check to see if a camera fade is available on.</param>
+        /// <returns>Returns true if the headset has fade functionality on it.</returns>
+        public override bool HasHeadsetFade(Transform obj)
+        {
+            if (obj.GetComponentInChildren<VRTK_ScreenFade>())
+            {
+                return true;
+            }
+            return false;
+        }
+
+        /// <summary>
+        /// The AddHeadsetFade method attempts to add the fade functionality to the game object with the camera on it.
+        /// </summary>
+        /// <param name="camera">The Transform to with the camera on to add the fade functionality to.</param>
+        public override void AddHeadsetFade(Transform camera)
+        {
+            if (camera && !camera.GetComponent<VRTK_ScreenFade>())
```

Comment:
make the `camera` null comparison explicit so:

`if (camera != null && !camera.GetComponent<VRTK_ScreenFade>())`